From eaf0064767b9f463b620c36530212004ef3e4d79 Mon Sep 17 00:00:00 2001
From: Winson Chung <winsonc@google.com>
Date: Wed, 18 Oct 2017 13:49:42 -0700
Subject: [PATCH 007/306] Fix NPE when getting managed profile info.

- There can be a race condition where the cached profile user ids
  aren't yet updated but the actual user has already been removed.
  Ignore these users when computing the quiet profile users.

Bug: 67932220
Test: Add/remove a work profile and launch an app
Change-Id: I8888d30e431ca4ffdacc28cc15a80a2deaa23d10
(cherry picked from commit 129771a08fa29302af2b491c8c81779becb936c3)
---
 services/core/java/com/android/server/am/RecentTasks.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/RecentTasks.java b/services/core/java/com/android/server/am/RecentTasks.java
index 78274bdc0e0..ed3f5033a04 100644
--- a/services/core/java/com/android/server/am/RecentTasks.java
+++ b/services/core/java/com/android/server/am/RecentTasks.java
@@ -914,7 +914,7 @@ class RecentTasks {
         mTmpQuietProfileUserIds.clear();
         for (int userId : profileUserIds) {
             final UserInfo userInfo = mUserController.getUserInfo(userId);
-            if (userInfo.isManagedProfile() && userInfo.isQuietModeEnabled()) {
+            if (userInfo != null && userInfo.isManagedProfile() && userInfo.isQuietModeEnabled()) {
                 mTmpQuietProfileUserIds.put(userId, true);
             }
             if (DEBUG_RECENTS_TRIM_TASKS) Slog.d(TAG, "User: " + userInfo
-- 
2.17.1

